server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java

```diff
@@ -226,12 +226,22 @@ public BootstrapCheckResult check(BootstrapContext context) {
             final long initialHeapSize = getInitialHeapSize();
             final long maxHeapSize = getMaxHeapSize();
             if (initialHeapSize != 0 && maxHeapSize != 0 && initialHeapSize != maxHeapSize) {
-                final String message = String.format(
+                final String message;
+                if (isMemoryLocked()) {
+                    message = String.format(
+                        Locale.ROOT,
+                        "initial heap size [%d] not equal to maximum heap size [%d]; " +
+                            "this can cause resize pauses",
+                        getInitialHeapSize(),
+                        getMaxHeapSize());
+                } else {
+                    message = String.format(
                         Locale.ROOT,
                         "initial heap size [%d] not equal to maximum heap size [%d]; " +
-                                "this can cause resize pauses and prevents mlockall from locking the entire heap",
+                            "this can cause resize pauses and prevents memory locking from locking the entire heap",
```

Comment:
Shouldn't the memory locking part be in the other condition's message, since this is the *not* memory locked case?